### ISSUE TITLE
Support for partial updates

### ISSUE THREAD

**Issue by lehotskysamuel** [micronaut-projects/micronaut-data#131]
Hi,

I'd like to request a feature. While query methods are pretty powerful in M1 already, updates are still somewhat limited. We use JDBC and would like to be able to generate partial update statements with Micronaut Data.

Imagine I have an Account entity with 3 fields: id, firstName and lastName. Currently it is only possible to generate a full update method:
```void update(@Id Long id, String firstName, String lastName);```

It would be nice if it was possible to have methods like:
```
void updateFirstName(@Id Long id, String firstName);
void updateLastName(@Id Long id, String lastName);
```

This feature would be especially useful for us who don't use JPA, since the dirty checking isn't supported with JDBC.

**Comment by graemerocher** [comment]
Hmm.. `void updateFirstName(@Id Long id, String firstName);` doesn't work already? that surprises me

**Comment by lehotskysamuel** [comment]
Hmm, I'm sorry, you're right. I've just tried the code and it worked. In that case you can close the ticket.
I got confused by this comment in the docs, maybe the real bugfix would be to expand the documentation? :)

Section 5 Updating Data

Alternatively you can also define a method that features parameter names that match the properties of the entity name:

```Book persist(String title, int pages);```

In this case you must specify parameters **for all properties** other than those that are declared as @Nullable or as a @GeneratedValue, if you do not a compilation error will occur.

**Comment by graemerocher** [comment]
Thanks for confirming, PRs to the documentation are welcome

**Comment by PhuBaDuong** [comment]
Hi everyone, 

I've tried to update status field by using list of UUIDs. The  name was suggested by IDE (Intellij) 

Completable updateStatusByUuidInList(List<UUID> uuid, String status);

But at compile time I got this error

**Unable to implement Repository method: Phone.updateStatusByUuidInList(List uuid,String status). At least one parameter required to update**


Does anyone have any idea? Thank you so much!!
